src/datasource.ts

```diff
@@ -383,8 +450,45 @@ export default class CogniteDatasource {
     };
   }
 
-  annotationQuery(options: any) {
-    throw new Error("Annotation Support not implemented yet.");
+  public async annotationQuery(options: AnnotationQueryOptions) {
+    const { range, annotation } = options;
+    const { expr, filter, error } = annotation;
+    const startTime = Math.ceil(dateMath.parse(range.from));
+    const endTime = Math.ceil(dateMath.parse(range.to));
+    if (annotation.error) return [];
+
+    const queryOptions = this.parse(expr,ParseType.Event);
+    const filterOptions = this.parse(filter || '',ParseType.Event);
+
+    // need to have just equality
+    const equalCheck = queryOptions.filters.find(x => x.type !== '=');
+    if (equalCheck) {
+      return [{value: "ERROR: Query can only use '='"}];
+    }
+
+    let url = this.url + `/cogniteapi/${this.project}/events/search?`;
+    for (let param of queryOptions.filters) {
+      url += param.property + '=' + param.value + "&";
```

Comment:
We should use a `querystring` library is similarly so we don't have to construct these urls